scrapy/contracts/__init__.py

```diff
@@ -51,8 +51,11 @@ def from_method(self, method, results):
             # calculate request args
             args, kwargs = get_spec(Request.__init__)
             kwargs['callback'] = method
+            ignored_errors = []
             for contract in contracts:
                 kwargs = contract.adjust_request_args(kwargs)
+                if contract.name == 'ignore':
```

Comment:
Hm, now I'm not sure this is the way to go - for all other contract features we're calling Contract methods, not checking its name. What about adding `ignored_errors` attribute or `get_ignored_error_names()` function to base Contract class and overriding it in IgnoreContract? Or maybe it is possible to make it more powerful, by allowing a contract to decide if an exception should be ignored or not (`contract.should_ignore_error(...)` called when exception happens - it will be able to check exception details to decide if it should be ignored)? Am I over-thinking it?